src/actions/guides/database/pagination.cr

```diff
@@ -0,0 +1,422 @@
+    or `BrowserAction`. That way any action that inherits from them has access to the included methods:
+
+    ```crystal
+    # src/actions/browser_action.cr
+    abstract class BrowserAction < Lucky::Action
+      include Lucky::Paginator::BackendHelpers
+    end
+    ```
+
+    > We are using `BrowserAction` in this example, but it works the same for
+    ApiAction > or any other `Lucky::Action`
+
+    This module will add the `paginate` method to you actions. This is what
+    you'll use to paginate your query results.
+
+    ## Paginating
+
+    Pagination is easy-peasy, call `paginate` with a query:
+
+    ```crystal
+    class Users::Index < BrowserAction
+      get "/users" do
+        pages, users = paginate(UserQuery.new)
+        html IndexPages, users: users, pages: pages
+      end
+    end
+    ```
+
+    In this case `pages` is a `Lucky::Paginator` with metadata about the
+    current page, total numbers of pages, and helpers for generating paths to
+    next and previous pages.
+
+    The other variable, `users`, is the passed in query (`UserQuery.new`)
+    with a `limit` and `offset` applied so it returns records for just
+    the requested page.
+
+    `paginate` will use the `page` param or fallback to page `1` if `page` is
+    not present. The `page` param can be a query param, a body param, or a
+    route param.
+
+    Next we'll go over how to customize the defaults.
+
+    ## Customizing defaults
+
+    By default `paginate` returns `25` records per page and uses the `page`
+    param to determine which page is requested. But if you want something else,
+    you can!
+
+    ### Customizing records per page
+
+    There are 2 ways to customize items per page.
+
+    You can pass it to the `paginate` method:
+
+    ```crystal
+    paginate(UserQuery.new, per_page: 50)
+    ```
+
+    Or you can customize it with a method on your action:
+
+    ```crystal
+    abstract class BrowserAction < Lucky::Action
+      include Lucky::Paginator::BackendHelpers
+
+      def paginator_per_page : Int32
+        50 # Return a static value
+
+        # Or allow using a param with a default
+        params.get?(:per_page) || 50
+      end
+    end
+    ```
+
+    As you can see you can customize this however you want. You can use a
+    request headers, static value, param, or whatever custom logic you prefer.
+
+    ### Customize current page
+
+    By default `paginate` uses the `page` param. The param can come from a
+    query param, route param, or body param. If the `page` param is not
+    there, it will fall back to page `1`.
+
+    If this logic is not what you want, you can customize it by creating
+    a `paginator_page` that returns an `Int32`
+
+    ```crystal
+    abstract class ApiAction < Lucky::Action
+      include Lucky::Paginator::BackendHelpers
+
+      def paginator_page : Int32
+        request.headers["Page"]? || 1 # Use a header
```

Comment:
Would `request.headers["Page"]` return a String?

Reply:
Whoops! Yeah need to to_i it